bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckablePullRequest.java

```diff
@@ -200,12 +200,18 @@ PullRequestCheckIssueVisitor createVisitor() throws IOException {
         return new PullRequestCheckIssueVisitor(checks);
     }
 
-    void executeChecks(Hash localHash, CensusInstance censusInstance, PullRequestCheckIssueVisitor visitor, List<String> additionalConfiguration) throws IOException {
+    PullRequestCheckIssueVisitor createVisitorUsingHeadHash() throws IOException {
+        var checks = JCheck.checksFor(localRepo, pr.headHash());
+        return new PullRequestCheckIssueVisitor(checks);
+    }
+
+    void executeChecks(Hash localHash, CensusInstance censusInstance, PullRequestCheckIssueVisitor visitor,
+                       List<String> additionalConfiguration, boolean useTargetConf) throws IOException {
```

Comment:
Instead of a boolean `useTargetConf`, I think it's better if we just supply the hash we want to use (target or src, if confOverride is set, then this method will handle it).

Reply:
Yeah, pass hash is better. But even if we pass hash here, it won't affect the logic of using confOverride because confOverride is set in the constructor of `CheckablePullRequest`

Reply:
Now, I prefer not to change this method because I think ` CheckablePullRequest#targetHash` can help us reuse the target hash. If we add hash as an argument, we will need to call `PullRequestUtils.targetHash(localRepo)` more than once.

Reply:
`CheckablePullRequest#targetHash` is public so you can call it from `CheckRun`.

Reply:
Oh, then I will change the method. Thanks.